website/content/docs/connect/transparent-proxy.mdx

```diff
@@ -149,3 +153,16 @@ it can dial `sample-app.default.svc.cluster.local`, using
 If ACLs with default "deny" policy are enabled, it also needs a
 [ServiceIntention](/docs/connect/config-entries/service-intentions) allowing it to talk to
 `sample-app`.
+
+### Headless Services
+For services that are not addressed using a virtual cluster IP, the upstream service must be
+configured using the [DialedDirectly](/docs/connect/config-entries/service-defaults#dialeddirectly) 
+option. 
+
+Individual instance addresses can then be discovered using DNS, and dialed through the transparent proxy.
+When this mode is enabled on the upstream, connect certificates will be presented for mTLS and
+intentions will be enforced at the destination.
+
+Note that when dialing individual instances HTTP routing rules configured with config entries 
+will **not** be considered. The transparent proxy acts as a TCP proxy to the original
+destination IP address.
```

Comment:
This is the main bit that gives me pause about the current implementation.

Configuration like retrying requests is applied on HTTP routes, and currently the network filter for passthrough is a TCP proxy. If we did attach an HTTP connection manager, what route config would it use? If it used the one for the upstream _name_, we would no longer route to the passthrough cluster and would instead load balance across the route destination endpoints.

I'm not sure that we can support L7 rules for headless services without creating a separate service registration for each upstream address. This seems really limiting.